src/utils/io/mmap.rs

```diff
@@ -36,6 +40,49 @@ impl Size for Mmap {
     }
 }
 
+pub enum EitherMmapOrRandomAccessFile {
+    Mmap(Mmap),
+    RandomAccessFile(RandomAccessFile),
+}
+
+impl EitherMmapOrRandomAccessFile {
+    pub fn open(path: impl AsRef<Path>) -> io::Result<Self> {
+        Ok(if prefer_file_io_over_mmap() {
+            Self::RandomAccessFile(RandomAccessFile::open(path)?)
+        } else {
+            Self::Mmap(Mmap::map_path(path)?)
+        })
+    }
+}
+
+impl ReadAt for EitherMmapOrRandomAccessFile {
+    fn read_at(&self, pos: u64, buf: &mut [u8]) -> io::Result<usize> {
+        use EitherMmapOrRandomAccessFile::*;
+        match self {
+            Mmap(mmap) => mmap.read_at(pos, buf),
+            RandomAccessFile(file) => file.read_at(pos, buf),
+        }
+    }
+}
+
+impl Size for EitherMmapOrRandomAccessFile {
+    fn size(&self) -> io::Result<Option<u64>> {
+        use EitherMmapOrRandomAccessFile::*;
+        match self {
+            Mmap(mmap) => mmap.size(),
+            RandomAccessFile(file) => file.size(),
+        }
+    }
+}
+
+fn prefer_file_io_over_mmap() -> bool {
```

Comment:
A preference is optional. This is not optional. It's a requirement. If the environment variables tells us not to use mmap then we must not use mmap.

Reply:
I don't quite get it, could you elaborate?

Reply:
The name `prefer_file_io_over_mmap` implies that it's still optional whether we use mmap or file-io. But it is not optional. If we're told to use file-io then we must use file-io.

Reply:
How about renaming it to `should_use_file_io`?